libcxx/test/support/increasing_allocator.h

```diff
@@ -0,0 +1,43 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef INCREASING_ALLOCATOR_H
+#define INCREASING_ALLOCATOR_H
+
+#include <cstddef>
+#include <memory>
+
+#include "test_macros.h"
+
+#if TEST_STD_VER >= 23
+template <typename T>
+struct increasing_allocator {
```

Comment:
Could you please add a short comment explaining what this allocator does and in what ways it is special? Since this is promoted to a reusable utility, I think it's worth it.

Reply:
Thank you for your suggestion! I have added a short comment explaining the purpose and special functionality of the `increasing_allocator`. Please let me know if you have further suggestions on it. 

I have also modified the PR description so that it reflects the real purpose of this PR.